sycl/doc/extensions/proposed/sycl_ext_oneapi_prefetch.asciidoc

```diff
@@ -267,6 +280,33 @@ _Preconditions_: `ptr` must point to an object in global memory.
 
 _Effects_: Equivalent to `prefetch((void*) ptr, count * sizeof(T), properties)`.
 
+[source,c++]
+----
+template <typename T,
+          typename PtrProperties,
+          typename ArgProperties = ext::oneapi::experimental::properties<>>
+void prefetch(annotated_ptr<T, PtrProperties> ptr, ArgProperties properties = {});
+----
+_Preconditions_: `ptr` must point to an object in global memory.
+
+_Effects_: Equivalent to `prefetch((void*) ptr.get(), sizeof(T), properties)`
+
+[source,c++]
+----
+template <typename T,
+          typename PtrProperties,
+          typename ArgProperties = ext::oneapi::experimental::properties<>>
+void prefetch(annotated_ptr<T, PtrProperties> ptr, size_t count, ArgProperties properties = {});
+----
+_Preconditions_: `ptr` must point to an object in global memory.
+
+_Effects_: Equivalent to `prefetch((void*) ptr.get(), count * sizeof(T), properties)`
+
+[NOTE]
+====
+The current version of the above functions effectively ignores all annotations in the `annotated_ptr` when issuing a prefetch. This behavior may change in future, if other extensions introduce properties that interact with or modify prefetch behavior.
```

Comment:
This seems dangerous to me, especially with the default value of `properties`.  Consider code like this:

```
float *p = /*...*/;
annotated_ptr ap{p, properties{cache_control_read_cached<cache_level::L2>};
prefetch(ap);
```

Wouldn't the user expect this to prefetch into the L2 cache?  However, as currently worded, the property on the annotated pointer would be ignored.  Since there is no second parameter to `prefetch`, you get the empty set of properties, which is defined to prefetch into L1.

I'm wondering why we need these overloads at all.  Is it so hard for the user to call `.get` on their annotated pointer?  Forcing them to do this make it very clear in the code that the properties in the annotated pointer are ignored by the prefetch.